src/components/MultiHopTrade/hooks/useIsTradingActive.tsx

```diff
@@ -1,71 +1,59 @@
 import { selectInputBuyAsset, selectInputSellAsset } from 'state/slices/tradeInputSlice/selectors'
 import { selectActiveQuote, selectActiveSwapperName } from 'state/slices/tradeQuoteSlice/selectors'
-import { useAppDispatch, useAppSelector } from 'state/store'
+import { useAppSelector } from 'state/store'
 
 export const useIsTradingActive = () => {
   const activeQuote = useAppSelector(selectActiveQuote)
   const tradeType = (activeQuote as ThorEvmTradeQuote)?.tradeType
 
-  const [isTradingActiveOnSellPool, setIsTradingActiveOnSellPool] = useState(false)
-  const [isTradingActiveOnBuyPool, setIsTradingActiveOnBuyPool] = useState(false)
-
-  const dispatch = useAppDispatch()
-
   const buyAssetId = useAppSelector(selectInputBuyAsset).assetId
   const sellAssetId = useAppSelector(selectInputSellAsset).assetId
 
-  const { getIsTradingActive } = swapperApi.endpoints
   const swapperName = useAppSelector(selectActiveSwapperName)
 
-  useEffect(() => {
-    ;(async () => {
-      const isTradingActiveOnSellPoolResult =
-        sellAssetId &&
-        swapperName &&
-        (
-          await dispatch(
-            getIsTradingActive.initiate({
-              assetId: sellAssetId,
-              swapperName,
-            }),
-          )
-        ).data
-
-      setIsTradingActiveOnSellPool(!!isTradingActiveOnSellPoolResult)
-    })()
-  }, [dispatch, getIsTradingActive, sellAssetId, swapperName])
-
-  useEffect(() => {
-    ;(async () => {
-      const isTradingActiveOnBuyPoolResult =
-        buyAssetId &&
-        swapperName &&
-        (
-          await dispatch(
-            getIsTradingActive.initiate({
-              assetId: buyAssetId,
-              swapperName,
-            }),
-          )
-        ).data
-
-      setIsTradingActiveOnBuyPool(!!isTradingActiveOnBuyPoolResult)
-    })()
-  }, [buyAssetId, dispatch, getIsTradingActive, swapperName])
+  const { data: isTradingActiveOnSellPool } = useQuery({
+    ...reactQueries.common.isTradingActive({
+      assetId: sellAssetId,
+      swapperName,
+    }),
+    enabled: !!swapperName,
+    // @lukemorales/query-key-factory only returns queryFn and queryKey - all others will be ignored in the returned object
+    // Go stale instantly
+    staleTime: 0,
+    // Never store queries in cache since we always want fresh data
+    gcTime: 0,
+    refetchOnWindowFocus: true,
+    refetchOnMount: true,
+    refetchInterval: 60_000,
+  })
+
+  const { data: isTradingActiveOnBuyPool } = useQuery({
+    ...reactQueries.common.isTradingActive({
+      assetId: buyAssetId,
+      swapperName,
+    }),
+    enabled: !!swapperName,
+    // @lukemorales/query-key-factory only returns queryFn and queryKey - all others will be ignored in the returned object
+    // Go stale instantly
+    // Never store queries in cache since we always want fresh data
+    refetchOnWindowFocus: true,
+    refetchOnMount: true,
+    refetchInterval: 60_000,
+  })
 
   return {
```

Comment:
Moved this back to draft as I think we can make this better now that this uses react-query:

I see two ways to do this, either:

1. This module query-related logic could be moved into a `queries/hooks` hook of sorts that will simply wrap `isTradingActive` with default args (and isn't related to swapper domain), then this module will still exist with the current path, and consume two calls of it
2. For conciseness and perf, another option is to use `useQueries()` here - which would enable parallelism

Regardless of the option chosen (most likely 2.) we'll now be able to benefit from loading states here, which were previously missing 🎉

Reply:
Reopening as this will be tackled on the PR on top of this one